### PR TITLE
Change Application Connector secret namespace to kyma-system

### PR DESCRIPTION
**Pull request by mvshao**
**Description**
This PR changes the default location from which Application Connector components retrieve secrets for applications because it is required to install all Kyma components to the `kyma-system` namespace.

Changes proposed in this pull request:

- application-gateway looks for application credentials in the `kyma-system` namespace (previously was `kyma-integration`)
- removing dependencies to `kyma-integration` namespace in Compass-Runtime-Agent and Central-Connectivity-Validator 

**Related issue(s)**
#15915

### CODE REVIEW

**Review by grego952 — DISMISSED** [review]

**Comment by akgalwas** [comment]
Compass Runtime agent should albo be modified as it contains reference to `kyma-integration`.

**Comment by mvshao** [comment]
> Compass Runtime agent should albo be modified as it contains reference to `kyma-integration`.

As part of the idea of creating small PRs, I focused only on changes to Application Gateway, but I can easily add changes to Compass Runtime Agent and Application Validator.

**Review by franpog859 — COMMENTED** [review]
Please, add also changes in the Compass Runtime Agent and Central Application Connectivity Validator. Without those changes, Central Application Gateway won't find any secret in `kyma-system` namespace, and nothing will work

File: tests/components/application-connector/resources/installation-config/mini-kyma-skr.yaml [hunk 1]
```diff
@@ -9,6 +9,6 @@ prerequisites:
 components:
   - name: "istio-resources"
   - name: "application-connector"
-    namespace: "kyma-integration"
+    namespace: "kyma-system"
   - name: "compass-runtime-agent"
     namespace: "compass-system"
```
Comment (franpog859) [inline tests/components/application-connector/resources/installation-config/mini-kyma-skr.yaml:14]: Please, remove the non-default namespaces here

File: resources/application-connector/templates/limit-range.yaml [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-  name: kyma-default
-  namespace: {{ .Values.global.integrationNamespace }}
-  labels:
-    release: {{ .Release.Name }}
-    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
-    app.kubernetes.io/name: {{ template "name" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-spec:
-  limits:
```
[the remark this answers is not in the document]
Comment (mvshao) [reply resources/application-connector/templates/limit-range.yaml:13]: `kyma-system` is a global namespace for whole Kyma components, and I think we shouldn't manipulate with its Limit Range. Linkt to .yaml Limit file -> https://github.com/kyma-project/kyma/blob/main/resources/cluster-essentials/templates/limit-range.yaml

File: components/central-application-gateway/README.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ Central Application Gateway has the following parameters:
 - **proxyPort** is the port that acts as a proxy for the calls from services and Functions to an external solution in the default standalone mode or Compass bundles with a single API definition. The default port is `8080`.
 - **proxyPortCompass** is the port that acts as a proxy for the calls from services and Functions to an external solution in the Compass mode. The default port is `8082`.
 - **externalAPIPort** is the port that exposes the API which allows checking the component status. The default port is `8081`.
-- **applicationSecretsNamespace** is the Namespace in which the Application secrets used by Central Application Gateway exist. The default Namespace is `kyma-integration`.
+- **applicationSecretsNamespace** is the Namespace in which the Application secrets used by Central Application Gateway exist. The default Namespace is `kyma-system`.
```
[the remark this answers is not in the document]
Comment (mvshao) [reply components/central-application-gateway/README.md:44]: Yes, having a migration script for secrets, on open-source, is a good idea, but we still need to discuss this approach. There is a high probability that Compass Runtime Agent will create duplicates in `kyma-system`. We need to investigate this scenario.

[the remark this answers is not in the document]
Comment (franpog859) [reply components/central-application-gateway/README.md:44]: Yup, I do think so too. The question is if Compass Runtime Agent applies all of the previous secrets to the `kyma-system` namespace so the migration script would just throw out the existing `kyma-integration` secrets or if there is a need to copy them

Another thing, the migration should be done probably before the upgrade process to enable `kyma-integration` namespace deletion. Is that correct?

[the remark this answers is not in the document]
Comment (mvshao) [reply components/central-application-gateway/README.md:44]: Yes, that is correct 😄, firstly we need to copy all the secrets from `kyma-integration` to `kyma-system`. The next step is to change the secret namespace to `kyma-system`.

**Review by franpog859 — COMMENTED** [review]
1. You should change also [this file](https://github.com/kyma-project/kyma/blob/main/installation/resources/components.yaml#L22).
2. Did you check the upgrade mentioned in [this comment](https://github.com/kyma-project/kyma/pull/16024/files#r1021466123)?

**Review by franpog859 — APPROVED** [review]

**Review by franpog859 — APPROVED** [review]

**Review by grego952 — DISMISSED** [review]

**Review by pbochynski — DISMISSED** [review]

**Review by franpog859 — APPROVED** [review]

**Review by tobiscr — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Review by franpog859 — APPROVED** [review]

**Review by tobiscr — APPROVED** [review]

**Review by pbochynski — APPROVED** [review]

**Pull request merged**